bugbot/rules/file_crash_bug.py

```diff
@@ -0,0 +1,156 @@
+                {
+                    **socorro_util.generate_bug_description_data(report),
+                    "signature": signature,
+                    "needinfo_regression_author": bool(flags),
+                }
+            )
+
+            # TODO: Provide the following information:
+            # [X] Crash signature
+            # [X] Top 10 frames of crashing thread
+            # [X] Component
+            # [X] The kind of crash
+            # [ ] Regression window
+            # [X] Inducing patch
+            # [X] Reason
+            # [X] Regressed by
+            # [X] Platform
+            # [ ] Firefox status flags
+            # [ ] Severity
+            # [ ] Time correlation
+            # [X] User comments
+            # [ ] Crash address commonalities
+            # [ ] Estimated future crash volume
+
+            bug_data = {
+                "blocks": "bugbot-auto-crash",
+                "type": "defect",
+                "keywords": ["crash"],
+                "status_whiteboard": f"[bugbot-crash-v{self.VERSION}]",
+                "summary": title,
+                "product": signature.crash_component.product,
+                "component": signature.crash_component.name,
+                "op_sys": signature.bugzilla_op_sys,
+                "rep_platform": signature.bugzilla_cpu_arch,
+                "cf_crash_signature": f"[@ {signature.signature_term}]",
+                "description": description,
+                # TODO: Uncomment the following lines when we move to file on
+                # the production instance of Bugzilla. Filling `regressed_by` or
+                # `flags` on bugzilla-dev will cause "bug does not exist" errors.
+                # "regressed_by": signature.regressed_by,
+                # "flags": flags,
+            }
+
+            if self.dryrun:
+                logger.info("Dry-run bug:")
+                pprint.pprint(bug_data)
+                bug_id = str(len(bugs) + 1)
+            else:
+                # NOTE: When moving to production:
+                #   - Use Bugzilla instead of DevBugzilla
+                #   - Drop the DevBugzilla class
+                #   - Update the bug URL `file_crash_bug.html`
+                #   - Drop the bug link `file_crash_bug_description.md.jinja`
+                #   - Fill the `regressed_by` and `flags` fields
+                #   - Create the bug using `utils.create_bug``
+                resp = requests.post(
+                    url=DevBugzilla.API_URL,
+                    json=bug_data,
+                    headers=DevBugzilla([]).get_header(),
+                    verify=True,
+                    timeout=DevBugzilla.TIMEOUT,
+                )
+                resp.raise_for_status()
+                bug = resp.json()
+                bug_id = str(bug["id"])
+                # TODO: log the created bugs info somewhere (e.g., DB,
+                # spreadsheet, or LabelStudio)
```

Comment:
We could have a separate script for this which simply lists the bugs blocking the meta bug

Reply:
I would like first to finalize what data we need to collect. If all data we need are on Bugzilla, then a separate script will be a cleaner solution.

Alternatively, we could create a solution that is usable by all tools. We have some outdated code that aims to do something like that: https://github.com/mozilla/bugbot/blob/master/bugbot/db.py.